component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java

```diff
@@ -414,45 +418,49 @@ private int parseGroupMembers(List<BaseObject> groupObjs, String type, XWikiCont
+          getContext().getLanguage());
       unsubscribeFooter = messageTool.get("cel_newsletter_unsubscribe_footer", Arrays.asList(
-          getUnsubscribeLink(blogDocument.getSpace(), emailAddress, context)));
+          getUnsubscribeLink(blogDocument.getSpace(), emailAddress)));
     }
     return unsubscribeFooter;
   }
 
-  private String getUnsubscribeLink(String blogSpace, String emailAddresse, XWikiContext context)
-      throws XWikiException {
+  String getUnsubscribeLink(String blogSpace, String emailAddresse) throws XWikiException {
     String unsubscribeLink = "";
-    XWikiDocument blogDocument = BlogUtils.getInstance().getBlogPageByBlogSpace(blogSpace, context);
-    BaseObject blogObj = blogDocument.getObject("Celements2.BlogConfigClass", false, context);
+    XWikiDocument blogDocument = BlogUtils.getInstance().getBlogPageByBlogSpace(blogSpace,
+        getContext());
+    BaseObject blogObj = blogDocument.getObject("Celements2.BlogConfigClass", false, getContext());
     if ((blogObj != null) && (blogObj.getIntValue("unsubscribe_info") == 1)) {
-      unsubscribeLink = blogDocument.getExternalURL("view",
-          "xpage=celements_ajax&ajax_mode=BlogAjax&doaction=unsubscribe" + "&emailadresse="
-              + emailAddresse, context);
+      try {
+        unsubscribeLink = blogDocument.getExternalURL("view",
+            "xpage=celements_ajax&ajax_mode=BlogAjax&doaction=unsubscribe&emailadresse="
+                + URLEncoder.encode(emailAddresse, UTF_8.name()), getContext());
+      } catch (UnsupportedEncodingException e) {
+        LOGGER.error("UTF-8 unsupported.");
+      }
     }
     return unsubscribeLink;
   }
 
-  private String getHtmlContent(XWikiDocument doc, String baseURL, XWikiContext context)
-      throws XWikiException {
+  private String getHtmlContent(XWikiDocument doc, String baseURL) throws XWikiException {
     String header = "";
     if ((baseURL != null) && !"".equals(baseURL.trim())) {
       header = "<base href='" + baseURL + "' />\n";
     }
-    String renderLang = context.getLanguage();
-    VelocityContext vcontext = (VelocityContext) context.get("vcontext");
+    String renderLang = getContext().getLanguage();
+    VelocityContext vcontext = (VelocityContext) getContext().get("vcontext");
     XWikiMessageTool msgTool = getWebUtilsService().getMessageTool(renderLang);
     DocumentReference headerRef = getWebUtilsService().resolveDocumentReference(
         "LocalMacros.NewsletterHTMLheader");
-    if (getContext().getWiki().exists(headerRef, context)) {
+    if (getContext().getWiki().exists(headerRef, getContext())) {
       LOGGER.debug("Additional header found.");
-      LOGGER.debug("doc=" + doc + ", context.language=" + context.getLanguage());
-      LOGGER.debug("context=" + context);
+      LOGGER.debug("doc=" + doc + ", context.language=" + getContext().getLanguage());
+      LOGGER.debug("context=" + getContext());
```

Comment:
Please use when logging like this following format: LOGGER.debug("doc={}, context.languag={}", doc, context)